maven/lib/dependabot/maven/update_checker/version_finder.rb

```diff
@@ -257,15 +258,16 @@ def dependency_metadata_url(repository_url)
         end
 
         def dependency_files_url(repository_url, version)
-          group_id, artifact_id = dependency.name.split(":")
+          group_id, artifact_id, classifier = dependency.name.split(":")
```

Comment:
Looks like this also needs to change in `dependency_metadata_url`

Reply:
Looks like there's also quite a few instances where we split and get the last element as the artifact id in the MetadataFinder, e.g. https://github.com/dependabot/dependabot-core/blob/master/maven/lib/dependabot/maven/metadata_finder.rb#L26

Reply:
@feelepxyz From what I can tell the classifier itself it not used in the [url for fetching maven metadata](https://github.com/dependabot/dependabot-core/blob/master/maven/lib/dependabot/maven/update_checker/version_finder.rb#L250).  Mulesoft dependencies use classifiers a lot, an example being [the pom file in this repo I created for a Mulesoft support ticket](https://github.com/a1flecke/mule-assert-equals-tester/blob/assert-equals/pom.xml#L58-L82).  I checked a [couple](https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/connectors/mule-http-connector/maven-metadata.xml) of the [metadata paths](https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/connectors/mule-jms-client/maven-metadata.xml) for those dependencies and only `group_id` and `artifact_id` were needed.

Thank you for catching the other `.last` calls.  I will fix those